File: 240531/함수를 이용한 합과 소수 판별/use-functions-to-determine-sums-and-decimals.py
def is_check_2(num):
    s = 0

    while num > 0:
        s += num % 10
        num //= 10
    if s%2 == 0:
        return True
    return False

def is_check(num):
    check = True
    if num == 1:
        return False
    
    for i in range(2, num):
        if num%i == 0:
            check = False
    return check and is_check_2(num)

a, b = tuple(map(int, input().split()))

cnt = 0

for num in range(a, b+1):
    if is_check(num):
        cnt += 1

print(cnt)

File: 241230/함수를 이용한 합과 소수 판별/use-functions-to-determine-sums-and-decimals.cpp
#include <iostream>
using namespace std;

bool F(int num) {
    bool check = true;
    for (int i=2; i<num; i++) {
        if(num%i == 0) {
            check = false;
            break;
        }
    }
    if (check) {
        int sum_val = 0;
        while (num > 0) {
            sum_val += num%10;
            num /= 10;
        }
        if (sum_val%2 == 0) {
            return true;
        } else {
            return false;
        }
    } else {
        return false;
    }
}

int main() {
    // 여기에 코드를 작성해주세요.
    int a, b, cnt=0;
    cin >> a >> b;
    for (int i=a; i<=b; i++) {
        if (F(i)) cnt++;
    }
    cout << cnt;
    return 0;
}